src/web/page/templates.rs

```diff
@@ -0,0 +1,191 @@
+use super::TEMPLATE_DATA;
+use crate::error::Result;
+use arc_swap::ArcSwap;
+use serde_json::Value;
+use std::collections::HashMap;
+use tera::{Result as TeraResult, Tera};
+
+/// Holds all data relevant to templating
+pub(crate) struct TemplateData {
+    /// The actual templates, stored in an `ArcSwap` so that they're hot-swappable
+    // TODO: Conditional compilation so it's not always wrapped, the `ArcSwap` is unneeded overhead for prod
+    pub templates: ArcSwap<Tera>,
+    /// The current global alert, serialized into a json value
+    global_alert: Value,
+    /// The version of docs.rs, serialized into a json value
+    docsrs_version: Value,
+    /// The current resource suffix of rustc, serialized into a json value
+    resource_suffix: Value,
```

Comment:
It'd be helpful to explain in a comment *why* those values are serialized into a `Value`.

When I started reviewing this PR I wondered by those were encoded as value instead of just using `#[derive(Serialize)]`, but then I noticed you exposed those values as Tera functions (which is quite clever!), and indeed it makes sense those are pre-serialized. Explaining it would aid understanding the code.